Solutions/Marain.UserNotifications.Abstractions/Marain/UserNotifications/UserNotification.cs

```diff
@@ -45,7 +46,7 @@ public class UserNotification
             this.Timestamp = timestamp != default ? timestamp.ToUniversalTime() : throw new ArgumentException(nameof(timestamp));
             this.Properties = properties ?? throw new ArgumentNullException(nameof(properties));
             this.Metadata = metadata ?? throw new ArgumentNullException(nameof(metadata));
-            this.ChannelDeliveryStatuses = channelDeliveryStatuses?.ToList() ?? new List<UserNotificationStatus>();
+            this.ChannelDeliveryStatuses = (channelDeliveryStatuses ?? Enumerable.Empty<UserNotificationStatus>()).ToImmutableArray();
```

Comment:
Immutable types! Fantastic! (I thought I'd be pushing it too far to suggest those, but I'm happy to see them.)

One thing: is the `null` case common here? I ask because I'm not sure of the efficiency of this code. I'm wondering about something more like this:

```csharp
this.ChannelDeliveryStatuses = channelDeliveryStatuses.?ToImmutableArray() ?? ImmutableArray<UserNotificationStatus>.Empty;
```

In the case where `null` gets passed, this does less work: it goes straight to returning an empty immutable array, instead of first obtaining an empty `IEnumerable<UserNotificationStatus>` that will eventually be turned into an empty immutable array. As it happens the immutable types are sufficiently clever here: they will detect that this is an empty array, and so will skip any attempt to enumerate it. But it still takes some work to detect that, and it still ends up allocating a brand new `ImmutableArray` instead of using the singleton empty one.